randovania/games/fusion/exporter/patch_data_factory.py

```diff
@@ -54,6 +54,8 @@ def _create_pickup_dict(self, pickup_list: list[ExportedPickupDetails]) -> dict:
                 is_major = True
             if not pickup.is_for_remote_player and pickup.conditional_resources[0].resources:
                 resource = pickup.conditional_resources[0].resources[-1][0].extra["item"]
+                jingle = pickup.conditional_resources[0].resources[-1][0].extra.get("Jingle", "Minor")
+                message_id = pickup.conditional_resources[0].resources[-1][0].extra.get("MessageID", None)
             else:
                 resource = "None"
 
```

Comment:
I would redo the logic, so that the "custom_message" var below gets treated as the dict for ItemMessages. So that later down the line you can still say "major_entry["ItemMessages"] = custom_messages without having weird conditions or building a different dict there. Otherwise this reads really weird.
Very recommended to rename it for that